src/src/com/microsoft/aad/adal/SerializedTokenCacheItem.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.aad.adal;
+
+import java.io.UnsupportedEncodingException;
+import org.json.JSONException;
+import com.google.gson.annotations.SerializedName;
+
+final class SerializedTokenCacheItem {
+
+    private static final String TAG = SerializedTokenCacheItem.class.getSimpleName();
+
+    @SerializedName("authority")
+    private String mAuthority;
+
+    @SerializedName("refreshToken")
+    private String mRefreshToken;
+
+    @SerializedName("idToken")
+    private String mIdToken;
+    
+    @SerializedName("familyClientId")
+    private String mFamilyClientId;
+    
+    SerializedTokenCacheItem(final TokenCacheItem tokenCacheItem) {
+        this.mAuthority = tokenCacheItem.getAuthority();
+        this.mRefreshToken = tokenCacheItem.getRefreshToken();
+        this.mIdToken = tokenCacheItem.getRawIdToken();
+        this.mFamilyClientId = tokenCacheItem.getFamilyClientId();        
+    }
+    
+    public TokenCacheItem parseSerializedTokenCacheItem() {
+        TokenCacheItem tokenCacheItem = new TokenCacheItem();
+        IdToken idToken;
+        
+        try {
+            idToken = IdToken.create(this.mIdToken);
+            UserInfo userInfo = new UserInfo(idToken);
+            tokenCacheItem.setUserInfo(userInfo);
+            tokenCacheItem.setTenantId(idToken.getTenantId());
+        } catch (UnsupportedEncodingException | JSONException e) {
```

Comment:
This catch seems targetted only at IdToken.create, make the scope tighter if true.